app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(errors[i])
+                   for i in range(len(errors))],
+        "path": file_path,
+        "status": "failed" if len(errors) > 0 else "passed"
```

Comment:
You can rewrite this condition as:
```suggestion
        "status": "failed" if errors else "passed"
```